doc/user-manual/language/cubical.lagda.rst

```diff
@@ -1,82 +1,145 @@
+    using ( isEquiv
+          ; equiv-proof
+          ; _≃_
+          ; primGlue )
+  open import Agda.Builtin.Sigma public
 
-  open import Agda.Builtin.Cubical.Path renaming (_≡_ to Path)
+  infix 2 Σ-syntax
 
-.. _cubical:
+  Σ-syntax : ∀ {ℓ ℓ'} (A : Set ℓ) (B : A → Set ℓ') → Set (ℓ ⊔ ℓ')
+  Σ-syntax = Σ
 
-***************************
-Cubical Type Theory in Agda
-***************************
+  syntax Σ-syntax A (λ x → B) = Σ[ x ∈ A ] B
 
-Cubical Type Theory `Cohen et al., Cubical`_ is implemented in Agda (beta version).
+  _×_ : ∀ {ℓ ℓ'} (A : Set ℓ) (B : Set ℓ') → Set (ℓ ⊔ ℓ')
+  A × B = Σ A (λ _ → B)
 
-To use cubical type theory, you need to run Agda with the ``--cubical`` command-line-option.
-You can also write ``{-# OPTIONS --cubical #-}`` at the top of the file.
+  infixr 5 _×_
 
-To define paths, use λ abstractions. For example, this is the definition of the constant path:
+.. _cubical:
 
-..
-  ::
-  module refl-example where
+*******
+Cubical
+*******
 
-::
+The Cubical mode extends Agda with a variety of features from Cubical
+Type Theory. In particular, computational univalence and higher
+inductive types which hence gives computational meaning to `Homotopy
+Type Theory and Univalent Foundations
+<https://homotopytypetheory.org/>`_. The version of Cubical Type
+Theory that Agda implements is a variation of the `CCHM`_ Cubical Type
+Theory where the Kan composition operations are decomposed into
+homogeneous composition and generalized transport. This is what makes
+the general schema for higher inductive types work, following the
+`CHM`_ paper.
+
+To use the cubical mode Agda needs to be run with the ``--cubical``
+command-line-option or with ``{-# OPTIONS --cubical #-}`` at the top
+of the file.
+
+The cubical mode adds the following features to Agda:
+
+1. An interval type and path types
+2. Generalized transport (``transp``)
+3. Partial elements and systems
+4. Homogeneous composition (``hcomp``)
+5. Glue types
+6. Higher inductive types
+7. Cubical identity types
 
-    refl : ∀ {a} {A : Set a} {x : A} → Path x x
-    refl {x = x} = λ i → x
+There is a standard ``agda/cubical`` library for Cubical Agda
+available at https://github.com/agda/cubical. This documentation uses
+the naming conventions of this library, for a detailed list of all of
+the built-in Cubical Agda files and primitives see
+:ref:`primitives-ref`. The main design choices of the core part of the
+library are explained in
+https://homotopytypetheory.org/2018/12/06/cubical-agda/.
 
-Although they use the same syntax, a path is not a function.
-For example, the following is not valid:
+The recommended way to get access to the Cubical primitives is to add
+the following to the top of a file (this assumes that the
+``agda/cubical`` library is installed and visible to Agda).
 
 .. code-block:: agda
 
-  refl : ∀ {a} {A : Set a} {x : A} → Path x x
-  refl {x = x} = λ (i : _) → x
+  {-# OPTIONS --cubical #-}
+  module MyModule where
 
```

Comment:
It’s actually optional to start files with `module MyModule where`, so rather than inventing a dummy name, you could just leave out that line.

Reply:
Oh, I didn't know. I'll remove it